sql: add a comment for unused SupportedVTables

yuzefovich opened the pull request:
It was added in b61e8348fad53899d46a806e1124073704ee7458 and then the single usage was removed in the same PR in 7b015d087ddffb70db74aeb632839a502368eb51. We want to keep this for code documentation purposes (and perhaps will use in the near future), so let's add a comment.

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/142429)
<!-- Reviewable:end -->

dt commented:
So these are the "supported in production" v-tables as listed by the public docs. I'd keep the set as we might want to make accessing non-supported vtables emit a NOTICE in the near future, and/or have access to them blocked entirely outside of non-internal, non-debugging sessions as outlined in this product brief: https://docs.google.com/document/d/1CR2_5ERATgHBXpQvKvetiPhTpNQWC506tSqrQKOb8Js/edit?tab=t.0#heading=h.8bkkp12botix

yuzefovich committed:
sql: add a comment for unused SupportedVTables

It was added in b61e8348fad53899d46a806e1124073704ee7458 and then
the single usage was removed in the same PR in
7b015d087ddffb70db74aeb632839a502368eb51. We want to keep this for
code documentation purposes (and perhaps will use in the near future),
so let's add a comment.

Release note: None

yuzefovich commented:
Ack, let's add a quick comment then. PTAL.

yuzefovich commented:
TFTR!

bors r+
